src/components/config/index.ts

```diff
@@ -82,34 +83,65 @@ const handshakeAndValidate = async (config: Config) => {
   }
 }
 
-export const setupConfigFromFile = async (path: string, watch: boolean) => {
+const fetchConfigAndValidate = async (path: string) => {
+  const fetched = await fetchConfig(path)
+  await handshakeAndValidate(fetched)
+  return fetched
+}
+
+const parseConfigAndValidate = async (path: string) => {
   const parsed = parseConfig(path)
   await handshakeAndValidate(parsed)
+  return parsed
+}
+
+export const setupConfigFromConfigFlag = async (
+  path: string,
+  watch: boolean,
+  checkingInterval: number
+) => {
+  if (isUrl(path)) {
+    const fetched = await fetchConfigAndValidate(path)
+    cfg = fetched
+    cfg.version = cfg.version || md5Hash(cfg)
+
+    setInterval(async () => {
+      const fetched = await fetchConfigAndValidate(path)
+      updateConfig(fetched)
+    }, checkingInterval * 1000)
+
+    return
+  }
+
+  const parsed = await parseConfigAndValidate(path)
   cfg = parsed
   cfg.version = cfg.version || md5Hash(cfg)
 
   if (watch) {
     const fileWatcher = chokidar.watch(path)
     fileWatcher.on('change', async () => {
-      const parsed = parseConfig(path)
-      await handshakeAndValidate(parsed)
+      const parsed = await parseConfigAndValidate(path)
       updateConfig(parsed)
     })
   }
 }
 
-export const setupConfigFromUrl = async (
-  url: string,
-  checkingInterval: number
-) => {
-  const fetched = await fetchConfig(url)
-  await handshakeAndValidate(fetched)
-  cfg = fetched
+const parseHarAndValidate = async (path: string) => {
+  const parsed = parseHarFile(path)
+  await handshakeAndValidate(parsed)
```

Comment:
I don't think we need to do "handshake" for the HAR file. Monika performs handshake with Symon if the config is specified in monika.json. There won't be such configuration in a HAR file.

Reply:
removed handhshake